lib/errors/error-handler.ts

```
@@ -0,0 +1,323 @@
+    
+    const errorsByCategory: Record<string, number> = {};
+    for (const [key, count] of this.errorCounts.entries()) {
+      const category = key.split(':')[0];
+      errorsByCategory[category] = (errorsByCategory[category] || 0) + count;
+    }
+
+    const recentErrors = Array.from(this.errorCounts.entries())
+      .map(([key, count]) => ({
+        key,
+        count,
+        lastOccurred: this.lastErrors.get(key)!,
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)